validator/client/grpc-api/grpc_validator_client_test.go

```diff
@@ -18,6 +19,58 @@ import (
 	"google.golang.org/protobuf/types/known/emptypb"
 )
 
+// toValidatorDutiesContainer is assumed to be available from your package, returning a *v1alpha1.ValidatorDutiesContainer.
+func TestToValidatorDutiesContainer_HappyPath(t *testing.T) {
+	// Create a mock DutiesResponse with current and next duties.
+	dutiesResp := &eth.DutiesResponse{
+		CurrentEpochDuties: []*eth.DutiesResponse_Duty{
+			{
+				Committee:        []primitives.ValidatorIndex{100, 101},
+				CommitteeIndex:   4,
+				AttesterSlot:     200,
+				ProposerSlots:    []primitives.Slot{400},
+				PublicKey:        []byte{0xAA, 0xBB},
+				Status:           eth.ValidatorStatus_ACTIVE,
+				ValidatorIndex:   101,
+				IsSyncCommittee:  false,
+				CommitteesAtSlot: 2,
+			},
+		},
+		NextEpochDuties: []*eth.DutiesResponse_Duty{
+			{
+				Committee:        []primitives.ValidatorIndex{300, 301},
+				CommitteeIndex:   8,
+				AttesterSlot:     600,
+				ProposerSlots:    []primitives.Slot{700, 701},
+				PublicKey:        []byte{0xCC, 0xDD},
+				Status:           eth.ValidatorStatus_ACTIVE,
+				ValidatorIndex:   301,
+				IsSyncCommittee:  true,
+				CommitteesAtSlot: 3,
+			},
+		},
+	}
+
+	gotContainer, err := toValidatorDutiesContainer(dutiesResp)
+	require.NoError(t, err)
+
+	// Validate we have the correct number of duties in current and next epochs.
+	require.Equal(t, len(gotContainer.CurrentEpochDuties), len(dutiesResp.CurrentEpochDuties))
+	require.Equal(t, len(gotContainer.NextEpochDuties), len(dutiesResp.NextEpochDuties))
+
+	firstCurrentDuty := gotContainer.CurrentEpochDuties[0]
+	expectedCurrentDuty := dutiesResp.CurrentEpochDuties[0]
+	require.DeepEqual(t, firstCurrentDuty.PublicKey, expectedCurrentDuty.PublicKey)
+	require.Equal(t, firstCurrentDuty.ValidatorIndex, expectedCurrentDuty.ValidatorIndex)
+	require.DeepEqual(t, firstCurrentDuty.ProposerSlots, expectedCurrentDuty.ProposerSlots)
+
+	firstNextDuty := gotContainer.NextEpochDuties[0]
+	expectedNextDuty := dutiesResp.NextEpochDuties[0]
+	require.DeepEqual(t, firstNextDuty.PublicKey, expectedNextDuty.PublicKey)
+	require.Equal(t, firstNextDuty.ValidatorIndex, expectedNextDuty.ValidatorIndex)
+	require.DeepEqual(t, firstNextDuty.ProposerSlots, expectedNextDuty.ProposerSlots)
```

Comment:
nitpick: I would convert `require` to `assert` because these checks are not critical